js/reference.js

```diff
@@ -2716,18 +2724,60 @@
             }
             // heuristics
             else {
+                // fetch config option for system columns heuristics (true|false|Array)
+                // if true, add all system columns
+                // if false, add none
+                // if array, add the ones defined
+
+                // order of system columns will always be the same
+                // RID will always be first in the visible columns list
+                // the rest will always be at the end ('RCB', 'RMB', 'RCT', 'RMT')
+
+                var systemColumnsMode = null;
+                // if compact or detailed, check for system column config option
+                if (module._isCompactContext(this._context)) {
```

Comment:
If you're sending the context to chaise, there's no reason to check the context here too. The same as `appLinkFn` send the context to that function and let that function deal with it.

Reply:
I know you've done this to simplify checking for sub-contexts of `compact`, but I feel like it's better to have all of these logic in one place. ermerstjs just needs to know that there's a callback that it can call.

Reply:
I wasn't sure of this when I was writing this. I saw that we had boolean checks similar to this in `ermrestJS` already, so mimic'ed that without comparing it to the applink function case.